File: CS-Examples/08_FilteringAndValidation/AutofilterNonBlank.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace AutofilterNonBlank
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\AutofilterBlank.xlsx");

            //Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            //Match the non blank data
            sheet.AutoFilters.MatchNonBlanks(0);

            //Filter
            sheet.AutoFilters.Filter();

            //Save the document
            string output = "AutofilterNonBlank_out.xlsx";
            workbook.SaveToFile(output,ExcelVersion.Version2013);

            //Launch the Excel file
            ExcelDocViewer(output);
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}

File: VB-Examples/08_FilteringAndValidation/AutofilterNonBlank.vb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace AutofilterNonBlank
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a workbook
			Dim workbook As New Workbook()

			'Load the document from disk
			workbook.LoadFromFile("..\..\..\..\..\..\Data\AutofilterBlank.xlsx")

			'Get the first worksheet
			Dim sheet As Worksheet = workbook.Worksheets(0)

			'Match the non blank data
			sheet.AutoFilters.MatchNonBlanks(0)

			'Filter
			sheet.AutoFilters.Filter()

			'Save the document
			Dim output As String = "AutofilterNonBlank_out.xlsx"
			workbook.SaveToFile(output,ExcelVersion.Version2013)

			'Launch the Excel file
			ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
